### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by bcasasus**
done safe-box exercice

### CODE REVIEW

File: staff/bernat-casasus/spotify-app/scripts/main.js [hunk 1]
```diff
@@ -23,13 +23,48 @@ search.onSearch(function (query) {
...
+                return {
+                    id: album.id,
+                    text: album.name
+                };
+            }));
+
+            detailContainer.clear();
+        })
+        .catch(function (error) {
+            alert('Sorry, we have temporary problem, try again later.');
```
Comment (SergioCastilloSorribes) [inline staff/bernat-casasus/spotify-app/scripts/main.js]: Tots els alerts del catch son iguals, no sabràs d'on ve l'error.

File: staff/bernat-casasus/spotify-app/scripts/logic.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 var logic = {
```
Comment (SergioCastilloSorribes) [inline staff/bernat-casasus/spotify-app/scripts/logic.js]: USE STRICT!!!!!!

File: staff/bernat-casasus/beers-app/scripts/logic.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+var logic = {
```
Comment (SergioCastilloSorribes) [inline staff/bernat-casasus/beers-app/scripts/logic.js]: USE STRICT

File: staff/bernat-casasus/restaurants-app.1/scripts/logic.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+var logic = {
+    searchBeers: function(query, callback) {
+        var request = new XMLHttpRequest();
+
+        request.onreadystatechange = function () {
+            if (request.readyState === 4 && request.status === 200) {
+                var beers = JSON.parse(request.responseText);
+
+                callback(beers);
```
Comment (alejaa03) [inline staff/bernat-casasus/restaurants-app.1/scripts/logic.js:9]: WTF BEERS IN RESTAURANT?!?!?!?!?!?!??!?!

File: staff/bernat-casasus/arrays/scripts/map.js [hunk 5]
```diff
@@ -1,3 +1,8 @@
 function map(array, func) {
     // TODO implement a map (it should work same to Array.prototype.map, but accepting the array as a parameter. use of Array.prototoype.map is forbidden)
+    var fulls = []
```
Comment (carlotavicens) [inline staff/bernat-casasus/arrays/scripts/map.js:3]: missed semicolon

File: staff/bernat-casasus/arrays/scripts/sort.js [hunk 6]
```diff
@@ -15,6 +15,28 @@ function sort(array, reverse) {
             }
         }
         return result;
+    }else{
+        if(reverse === false || reverse === undefined){
+
+        for(var i = 0; i < long; i++){
```
Comment (carlotavicens) [inline staff/bernat-casasus/arrays/scripts/sort.js]: tabula millor, aixi sabras on començar i on acaba una funció.

File: staff/bernat-casasus/react-tutorial/index.html [hunk 7]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Document</title>
```
Comment (carlotavicens) [inline staff/bernat-casasus/react-tutorial/index.html:7]: change the name

File: staff/bernat-casasus/recursive-for/index.html [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Document</title>
```
Comment (carlotavicens) [inline staff/bernat-casasus/recursive-for/index.html:7]: change the name

File: staff/bernat-casasus/restaurants-app.1/.versions/restaurants-app-1.0.0/scripts/web-components-1.0.0.js [hunk 9]
```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+/**
+ * Web Components lib
+ * 
+ * @author manuelbarzi
```
Comment (carlotavicens) [inline staff/bernat-casasus/restaurants-app.1/.versions/restaurants-app-1.0.0/scripts/web-components-1.0.0.js:6]: you are not the author of your files?

File: staff/bernat-casasus/restaurants-app.1/.versions/restaurants-app-1.1.0/scripts/assert-1.0.1.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+/**
+ * Asserts a condition (is true)
+ * 
+ * @param {string} message The description of the assertion
+ * @param {boolean} condition The condition to checked during assertion
+ * 
+ * @author manuelbarzi
```
Comment (carlotavicens) [inline staff/bernat-casasus/restaurants-app.1/.versions/restaurants-app-1.1.0/scripts/assert-1.0.1.js:9]: tu no ets el autor del teu fitxer?

File: staff/bernat-casasus/restaurants-app.1/.versions/restaurants-app-1.1.0/scripts/web-components-1.0.0.js [hunk 11]
```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+/**
+ * Web Components lib
+ * 
+ * @author manuelbarzi
```
Comment (carlotavicens) [inline staff/bernat-casasus/restaurants-app.1/.versions/restaurants-app-1.1.0/scripts/web-components-1.0.0.js:6]: tu no ets l'autor del teu arxiu?

File: staff/bernat-casasus/restaurants-app.1/index.html [hunk 12]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Document</title>
```
Comment (carlotavicens) [inline staff/bernat-casasus/restaurants-app.1/index.html:7]: canviar el titol

File: staff/bernat-casasus/restaurants-app.1/scripts/assert-1.0.1.js [hunk 13]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+/**
+ * Asserts a condition (is true)
+ * 
+ * @param {string} message The description of the assertion
+ * @param {boolean} condition The condition to checked during assertion
+ * 
+ * @author manuelbarzi
```
Comment (carlotavicens) [inline staff/bernat-casasus/restaurants-app.1/scripts/assert-1.0.1.js:9]: tu no ets l'autor dels teus fitxers¿

File: staff/bernat-casasus/restaurants-app.1/scripts/logic.js [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+        };
+
+        var url = 'https://quiet-inlet-67115.herokuapp.com/api/search/all?q=' + query;
+
+        request.open('get', url);
+
+        request.send();
+    },
+
+    retrieveBeerById: function(id, callback) {
```
Comment (carlotavicens) [inline staff/bernat-casasus/restaurants-app.1/scripts/logic.js:20]: beer quan es restaurant?

File: staff/bernat-casasus/restaurants-app.1/scripts/logic.js [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+        var request = new XMLHttpRequest();
+
+        request.onreadystatechange = function () {
+            if (request.readyState === 4 && request.status === 200) {
+                var beer = JSON.parse(request.responseText);
+
+                callback(beer);
```
Comment (carlotavicens) [inline staff/bernat-casasus/restaurants-app.1/scripts/logic.js:27]: beer in restaurant again

File: staff/bernat-casasus/restaurants-app.1/scripts/main.js [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+
+SearchPanel.prototype = Object.create(Component.prototype);
+SearchPanel.prototype.constructor = SearchPanel;
+
+var search = new SearchPanel();
+
+
+search.onSearch(function (query) {
+    
+    logic.searchBeers(query, function(beer){
```
Comment (carlotavicens) [inline staff/bernat-casasus/restaurants-app.1/scripts/main.js:38]: beer in restaurant again

File: staff/bernat-casasus/restaurants-app.1/scripts/main.js [hunk 17]
```diff
@@ -0,0 +1,145 @@
...
+
+var search = new SearchPanel();
+
+
+search.onSearch(function (query) {
+    
+    logic.searchBeers(query, function(beer){ 
+
+
+        results.updateResults(beer.map(function (result) {
```
Comment (carlotavicens) [inline staff/bernat-casasus/restaurants-app.1/scripts/main.js:41]: beer in restaurant again

File: staff/bernat-casasus/restaurants-app.1/scripts/main.js [hunk 18]
```diff
@@ -0,0 +1,145 @@
...
+};
+
+var results = new ResultsList();
+
+results.onItemClick(function (id, text) {
+
+    logic.retrieveBeerById(id, function(beer){ 
+
+
+        var detail = new DetailPanel(beer.name,beer.description, beer.labels.medium);
```
Comment (carlotavicens) [inline staff/bernat-casasus/restaurants-app.1/scripts/main.js:93]: beer in restaurant again

File: staff/bernat-casasus/restaurants-app.1/scripts/web-components-1.0.0.js [hunk 19]
```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+/**
+ * Web Components lib
+ * 
+ * @author manuelbarzi
```
Comment (carlotavicens) [inline staff/bernat-casasus/restaurants-app.1/scripts/web-components-1.0.0.js:6]: you are not the author of your files?

File: staff/bernat-casasus/restaurants-app.1/test.html [hunk 20]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Document</title>
```
Comment (carlotavicens) [inline staff/bernat-casasus/restaurants-app.1/test.html:7]: wrong title

File: staff/bernat-casasus/restaurants-app/.versions/restaurants-app-1.0.0/index.html [hunk 21]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Document</title>
```
Comment (carlotavicens) [inline staff/bernat-casasus/restaurants-app/.versions/restaurants-app-1.0.0/index.html:7]: the title!

File: staff/bernat-casasus/arrays/scripts/map.js (hunk 5 above)
Comment (bcasasus) [reply staff/bernat-casasus/arrays/scripts/map.js:3]: its ECMA6

File: staff/bernat-casasus/arrays/scripts/sort.js (hunk 6 above)
Comment (bcasasus) [reply staff/bernat-casasus/arrays/scripts/sort.js]: Yeah I will fix it with Alt+Shift+F shortcut in VS

File: staff/bernat-casasus/recursive-for/index.html (hunk 8 above)
Comment (bcasasus) [reply staff/bernat-casasus/recursive-for/index.html:7]: Please If posible use english to send comments! I will fix it!

File: staff/bernat-casasus/restaurants-app.1/.versions/restaurants-app-1.0.0/scripts/web-components-1.0.0.js (hunk 9 above)
Comment (bcasasus) [reply staff/bernat-casasus/restaurants-app.1/.versions/restaurants-app-1.0.0/scripts/web-components-1.0.0.js:6]: Please If posible use english to send comments!

File: staff/bernat-casasus/restaurants-app.1/scripts/main.js (hunk 17 above)
Comment (bcasasus) [reply staff/bernat-casasus/restaurants-app.1/scripts/main.js:41]: Yeah it is the testing version.

File: staff/bernat-casasus/restaurants-app.1/scripts/web-components-1.0.0.js (hunk 19 above)
Comment (bcasasus) [reply staff/bernat-casasus/restaurants-app.1/scripts/web-components-1.0.0.js:6]: Please If posible use english to send comments!

File: staff/bernat-casasus/restaurants-app.1/test.html (hunk 20 above)
Comment (bcasasus) [reply staff/bernat-casasus/restaurants-app.1/test.html:7]: Please If posible use english to send comments!

File: staff/bernat-casasus/restaurants-app/.versions/restaurants-app-1.0.0/index.html (hunk 21 above)
Comment (bcasasus) [reply staff/bernat-casasus/restaurants-app/.versions/restaurants-app-1.0.0/index.html:7]: What is "el titol" ? Please If posible use english to send comments!

**Pull request merged**